### PR TITLE
Enable registry mirror config for bottlerocket

### PR DESCRIPTION
**Pull request by abhinavmpandey08**
*Issue #, if available:*
#98

*Description of changes:*
Enable registry mirror config for bottlerocket
To configure registry mirror, add the following to the clusterconfig:
```yaml
  registryMirrorConfiguration:
    endpoint: <registry-mirror-endpoint>
    caCertContent: |
      -----BEGIN CERTIFICATE-----
      veryCoolCertificateData
      -----END CERTIFICATE-----
```
If you don't want to add the cert data in the clusterconfig, you can also set this env var
```
export EKSA_REGISTRY_MIRROR_CA="/path/to/cert"
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/api/v1alpha1/cluster.go
Comment (mrajashree) [inline pkg/api/v1alpha1/cluster.go:390]: does this mean if user has provided the [caCertContent](https://github.com/aws/eks-anywhere/blob/214151dfd037f55436ed73d0687a97f2bd5b161e/pkg/api/v1alpha1/cluster_types.go#L53), the CLI will override it with the env var's value? I think it should be the other way where if this field's value isn't provided in the cluster spec then we default to the env var

Comment (abhinavmpandey08) [reply pkg/api/v1alpha1/cluster.go:390]: I based the precedence order similar to how [aws cli](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html) does it. If a user specifies the env var `EKSA_REGISTRY_MIRROR_CA`, it will override the `caCertContent` in the config file. I believe capi/clusterctl does it this way as well

Comment (mrajashree) [reply pkg/api/v1alpha1/cluster.go:390]: @abhinavmpandey08 as per the [aws cli doc](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html) `If you specify an option by using a parameter on the AWS CLI command line, it overrides any value from either the corresponding environment variable or a profile in the configuration file.` To me, a user setting this file in the clusterspec is similar to passing a parameter for a particular cli command, so it should override any value from env var. what do you think?

Comment (abhinavmpandey08) [reply pkg/api/v1alpha1/cluster.go:390]: I don't mind either way. What you said makes sense. I will just update the logic

**Comment by abhay-krishna** [comment]
@abhinavmpandey08: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-cluster-controller-tooling-presubmit | 3b4d04701c6f93f4938484c64d51e8bb569a640d | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/275/eks-anywhere-cluster-controller-tooling-presubmit/1441052530510401536) | `/test eks-anywhere-cluster-controller-tooling-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=275). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:abhinavmpandey08).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by abhay-krishna** [comment]
@abhinavmpandey08: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-cluster-controller-tooling-presubmit | 3b4d04701c6f93f4938484c64d51e8bb569a640d | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/275/eks-anywhere-cluster-controller-tooling-presubmit/1441088071251529728) | `/test eks-anywhere-cluster-controller-tooling-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=275). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:abhinavmpandey08).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Review by mrajashree — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/275#" title="Author self-approved">abhinavmpandey08</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/275#pullrequestreview-762309912" title="Approved">mrajashree</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [abhinavmpandey08,mrajashree]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by abhay-krishna** [comment]
@abhinavmpandey08: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-cluster-controller-tooling-presubmit | 3b4d04701c6f93f4938484c64d51e8bb569a640d | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/275/eks-anywhere-cluster-controller-tooling-presubmit/1441088071251529728) | `/test eks-anywhere-cluster-controller-tooling-presubmit`
eks-anywhere-attribution-files-presubmit | e9875b5b0d8f5fedaac3fbf37c7370793c51a9d6 | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/275/eks-anywhere-attribution-files-presubmit/1441136328400965632) | `/test eks-anywhere-attribution-files-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=275). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:abhinavmpandey08).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Review by mdsgabriel — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/275#" title="Author self-approved">abhinavmpandey08</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/275#pullrequestreview-762489034" title="Approved">mdsgabriel</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/275#pullrequestreview-762309912" title="Approved">mrajashree</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [abhinavmpandey08,mdsgabriel,mrajashree]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by abhay-krishna** [comment]
@abhinavmpandey08: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-cluster-controller-tooling-presubmit | 3b4d04701c6f93f4938484c64d51e8bb569a640d | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/275/eks-anywhere-cluster-controller-tooling-presubmit/1441088071251529728) | `/test eks-anywhere-cluster-controller-tooling-presubmit`
eks-anywhere-attribution-files-presubmit | e9875b5b0d8f5fedaac3fbf37c7370793c51a9d6 | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/275/eks-anywhere-attribution-files-presubmit/1441148418247364608) | `/test eks-anywhere-attribution-files-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=275). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:abhinavmpandey08).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by abhay-krishna** [comment]
New changes are detected. LGTM label has been removed.

**Pull request merged**
